### ISSUE TITLE
Web - Emoji - Emoji picker is moved to the top when changing the size of the window

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51200]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.51-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** dave0123seife@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1, Navigate to any chat
2. Open emoji picker
3. Reduce the size of the window

## Expected Result:
Emoji picker stays in its position

## Actual Result:
Reducing the size of the windows cause the Emoji picker to be moved to the top

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/c1f0ebb0-e5c5-4820-ab2b-e07c70324a7d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@jliexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by jliexpensify** [comment]
@Expensify/design hello, is this a bug? I feel like it might be, as we're automatically moving it - but it could also be intentional, to get the picker out of the way as the window is now reduced. Keen for your thoughts, cheers!

**Comment by dubielzyk-expensify** [comment]
Yeah I would expect this to keep it's positioned anchored to the button that triggered it

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The emoji picker is moved to the top left when resizing window.

### What is the root cause of that problem?
In the emoji picker component, we already have a dimension listener to recalculate the position.
https://github.com/Expensify/App/blob/ac57d30444f5ed2bf0be3bc72705e0da64faa03e/src/components/EmojiPicker/EmojiPicker.tsx#L164-L173

However, the value returned is always 0 (-14 for top for me). The reason is, everytime we resize the window, the emoji picker button (which is the anchor where we calculate the position based on) is always re-rendered.
https://github.com/Expensify/App/blob/ac57d30444f5ed2bf0be3bc72705e0da64faa03e/src/components/EmojiPicker/EmojiPickerButton.tsx#L33-L45

I verified this by appending a random number to the ID and it keeps changing when resizing. So, when we are trying to recalculate the new position, the previous ref is not "valid" anymore. The emoji picker button keeps getting re-rendered because we pass inline function to the props.
https://github.com/Expensify/App/blob/ac57d30444f5ed2bf0be3bc72705e0da64faa03e/src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx#L518-L528

So, when the ReportActionCompose is re-rendered because of the window size, the emoji picker button is also re-rendered.

### What changes do you think we should make in order to solve the problem?

Wrap both function to a `useCallback` so it only re-renders when it's really need to.
https://github.com/Expensify/App/blob/ac57d30444f5ed2bf0be3bc72705e0da64faa03e/src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx#L518-L528

We need to apply this to other usage of emoji picker too, for example, in edit composer.

**Comment by shawnborton** [comment]
Same, agree this is a bug as I would also expect the emoji picker to stay anchored relative to the emoji button in the viewport.

**Comment by Anaslancer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The emoji picker is moved to the top left when resizing window.

### What is the root cause of that problem?
The `measureInWindow` function are not available until after the rendering has been completed in native.
https://github.com/Expensify/App/blob/96a68f871421afacceb3414e56a6437de462a949/src/libs/calculateAnchorPosition.ts#L15
By the way we are calling this function in during the rendering (when resize the window). So this function is returning the incorrect position (x:0, y:0).

### What changes do you think we should make in order to solve the problem?
We should call this function in `onLayout` at [here](https://github.com/Expensify/App/blob/96a68f871421afacceb3414e56a6437de462a949/src/components/EmojiPicker/EmojiPickerButton.tsx#L71).
For solving this error, I added a `moveEmojiPicker` function in below 2 files and called it in `onLayout`.
> [EmojiPicker.tsx](https://github.com/Expensify/App/blob/main/src/components/EmojiPicker/EmojiPicker.tsx)
> [EmojiPickerActions.ts](https://github.com/Expensify/App/blob/main/src/libs/actions/EmojiPickerAction.ts)

### What alternative solutions did you explore? (Optional)
N/A

### Contributor details
Your Expensify account email: anasup1995@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01aff093c9a804b145

**Comment by Anaslancer** [comment]
https://github.com/user-attachments/assets/85f20121-5a86-4588-8e5d-bb346407c76e

**Comment by Anaslancer** [comment]
Hi @abdulrahuman5196 Did you have a chance to review my [proposal](https://github.com/Expensify/App/issues/51200#issuecomment-2430047402) and [video](https://github.com/Expensify/App/issues/51200#issuecomment-2430054037)?

**Comment by jliexpensify** [comment]
Waiting on @abdulrahuman5196 to review

**Comment by raza-ak** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web - Emoji - Emoji picker is moved to the top when changing the size of the window
### What is the root cause of that problem?
The emoji picker position is controlled by a DOM reference (**emojiPickerRef**), which causes its position to reset and misalign when the window size changes.
### What changes do you think we should make in order to solve the problem?
To fix this, we implemented a custom CSS style to ensure the emoji picker stays fixed at the bottom-right corner. Additionally, we added a window resize handler using useEffect in the EmojiPickerButton component. This handler repositions the emoji picker to its designated bottom-right position every time the window is resized.

**Following changes will fix the issue:**
<img width="824" alt="image" src="https://github.com/user-attachments/assets/1469b8b4-0aed-4782-8967-bd5437941145">
<img width="747" alt="image" src="https://github.com/user-attachments/assets/4d6217b5-1e81-4f43-a38f-e192ccdff26b">
**Video:**

https://github.com/user-attachments/assets/2248668a-82c3-4567-b2ee-e601f207dc70

**Comment by abdulrahuman5196** [comment]
Hi, Will check on this today.

**Comment by abdulrahuman5196** [comment]
Checking now

**Comment by abdulrahuman5196** [comment]
I don't see this issue in latest dev. Checked Firefox, chrome and safari. @jliexpensify 


https://github.com/user-attachments/assets/d6b390a2-425e-4be6-ba1f-42a4a3b107ee

**Comment by jliexpensify** [comment]
Hmm yeah, it looks to be fixed in v9.0.55-0 now.

I was able to reproduce it once out of ~15 attempts (and only on the 1st attempt). Going to close this one then!
